src/app/core/services/repo-creator.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { flatMap } from 'rxjs/operators';
+import { Observable, of, pipe, UnaryFunction } from 'rxjs';
+import { GithubService } from './github.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RepoCreatorService {
+  constructor(
+    private githubService: GithubService
+  ) {}
+
+  /**
+   * Checks if the necessary repository has been created.
+   */
+  public verifyRepoCreation(phaseOwner: string, phaseRepo: string): UnaryFunction<Observable<boolean | null>, Observable<boolean>> {
+    return pipe(
+      flatMap((isFixAttempted: boolean | null) => {
+        if (!isFixAttempted) {
+          // If no fix has been attempted, there is no need to verify fix outcome.
+          return of(true);
+        } else if (isFixAttempted) {
+          // Verify that Repository has been created if a fix attempt has occurred.
+          return this.verifySessionAvailability(phaseOwner, phaseRepo);
+        }
+      })
+    );
+  }
+
+  /**
+   * Checks if the necessary repository is available and creates it if the permissions are available.
+   * @param phaseOwner Owner of Specified Repository.
+   * @param phaseRepo Name of Repository.
+   */
+  private verifySessionAvailability(phaseOwner: string, phaseRepo: string): Observable<boolean> {
+    return this.githubService.isRepositoryPresent(phaseOwner, phaseRepo);
+  }
```

Comment:
Since this is a private function that just encapsulates the `githubService` function with the same params should we just put the `githubService.isRepositoryPresent(...)` within `verifyRepoCreation` itself? Cuz your comment in Line 24 seems fairly clear already and would alr inform devs that the `githubService` function is being called to verify that a repo has been created.

Reply:
Fair enough, then in this case, we don't need to overuse abstraction and just put the method in there! :+1: